### ISSUE TITLE
Can't pull image via UI from on-prem Harbor OCI registry

### ISSUE THREAD

**Issue by jeffstoner** [containers/podman-desktop#2625]
### Bug description

I added our on-premise Harbor OCI registry and my credentials to Podman Desktop v1.0.1, then attempted to pull an image from it and got an authentication error.

### Operating system

Windows 10 Enterprise, Version 22H2, Build 19045.2965

### Version

1.0.x

### Steps to reproduce

Launch Podman Desktop
Go to Settings -> Registries, clicked `Add registry` button
entered url, username, password & clicked `Login` button

Go to Images, clicked `Pull an image` button
entered image url "registry.REDACTED/internal/imagename:1.1.0", clicked `Pull image` button

### Relevant log output

```shell
initializing source docker://registry.REDACTED/internal/imagename:1.1.0: reading manifest 1.1.0 in registry.REDACTED/internal/imagename: unauthorized: unauthorized to access repository: internal/imagename, action: pull: unauthorized to access repository: internal/imagename, action: pull
```

### Additional context

Pulled image successfully via command line (using Git Bash terminal):

```
$ podman.exe pull registry.REDACTED/internal/imagename:1.1.0
Trying to pull registry.REDACTED/internal/imagename:1.1.0...
Getting image source signatures
Copying blob sha256:33ddddad061bf7f7ac043a01e2d8b6d8df749017de959ac1cb80f1f7744d24a7
Copying blob sha256:7d06a89558206b8c7006ae6c6a9ee0f83f13a45281a1841ea40b1d4b2451fa7b
Copying blob sha256:55f11c11fb3e82387413c4441318bab7b01b86a6cdedabf22608b6b4a9bbb527
Copying blob sha256:d8dc2911810e8d8fa519bc23a22ad2157232eeecc4e24819528d256724d09dff
Copying blob sha256:39fbafb6c7ef40878b859fd0e4475c5ebd2ddfa8ed785a588594e8731b7cd0e8
Copying blob sha256:18f613c4a3d148d6367151e08930818c58cb5411d0668966c1e44805931af82c
Copying blob sha256:5f35e9fbba157006e806fb5c96fa0e49d094cf16302f59d96a45a2379fc39920
Copying config sha256:be01573c39f2e3a9fb02fcd0ae2798f38b22c4683b46215a0c07bd80ea84d569
Writing manifest to image destination
Storing signatures
be01573c39f2e3a9fb02fcd0ae2798f38b22c4683b46215a0c07bd80ea84d569
```

**Comment by benoitf** [comment]
thanks for the report @jeffstoner 

I've few questions:
When you added the registry through UI, you didn't get any error ?
The CLI command is working after you're doing the registration through the UI of Podman Desktop ?
Without adding the registry or removing registry in UI make the CLI command fails ?

If so, I guess Podman Desktop is not providing the right Auth parameters.

Do you reproduce the issue with demo.goharbor.io ?

**Comment by cdrage** [comment]
> demo.goharbor.io

Hi Jeff! I tried reproducing this to no avail. 

I tried using `demo.goharbor.io` creating an account and pulling `demo.goharbor.io/library/blackbox`:

![Screenshot 2023-06-20 at 2 52 26 PM](https://github.com/containers/podman-desktop/assets/6422176/df079841-bf7f-403b-a61e-93e9f9143042)

I also deployed my harbor install to a local kubernetes environment (using a lets encrypt tls being the only exception) to hopefully simulate the same corporate environment and I was also able to successfully pull the image.

Are you able to use the latest version of podman desktop + latest podman VM and try again? 

Looking at the code, the same `auth.json` is being used from the CLI as well as Podman Desktop so I see no reason why it should be failing. 

Let me know @jeffstoner
